src/app/public/modules/modal/modal-adapter.service.ts

```diff
@@ -11,24 +12,39 @@ export class SkyModalAdapterService {
 
   private docRef: any;
   private bodyEl: HTMLElement;
+  private renderer: Renderer2;
+  private hostRef: ComponentRef<SkyModalHostComponent>;
 
   constructor(
-    private windowRef: SkyWindowRefService) {
+    private rendererFactory: RendererFactory2,
+    private windowRef: SkyWindowRefService
+  ) {
+      this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
       this.docRef = this.windowRef.getWindow().document;
       this.bodyEl = this.windowRef.getWindow().document.body;
   }
 
   public addHostEl(): void {
-    this.bodyEl.appendChild(this.docRef.createElement('sky-modal-host'));
+    const domElem = (this.hostRef.hostView as EmbeddedViewRef<any>).rootNodes[0];
+
+    this.renderer.appendChild(this.bodyEl, domElem);
   }
 
   public removeHostEl(): void {
     const element = this.docRef.querySelector('sky-modal-host');
     if (element) {
-      this.bodyEl.removeChild(element);
+      this.renderer.removeChild(document.body, element);
```

Comment:
Replace `document.body` with `this.bodyEl`, so the window reference can be mocked out in a unit test.